tests/unit/amo/components/TestAddon.js

```diff
@@ -748,6 +771,242 @@ describe('Addon', () => {
       );
     });
   });
+
+  describe('more add-ons by authors', () => {
+    function mountMoreAddonsByAuthors({ addon = fakeAddon, ...others }) {
+      const root = mountRender({
+        addon,
+        ...others,
+      });
+
+      return root.find('.AddonDescription-more-addons');
+    }
+
+    it('fetches the other add-ons by authors', () => {
+      const addon = fakeAddon;
+      const fakeDispatch = sinon.stub();
+      shallowRender({ addon, dispatch: fakeDispatch });
+
+      sinon.assert.calledWith(fakeDispatch, fetchOtherAddonsByAuthors({
+        addonType: addon.type,
+        authors: addon.authors.map((author) => author.username),
+        errorHandlerId: createStubErrorHandler().id,
+        slug: addon.slug,
+      }));
+    });
+
+    it('does not fetch the other add-ons when add-on is the same', () => {
+      const addon = fakeAddon;
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      root.setProps({ addon });
+
+      sinon.assert.notCalled(fakeDispatch);
+    });
+
+    it('fetches the other add-ons when a new add-on is loaded', () => {
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon: fakeAddon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      const newAddon = { ...fakeAddon, slug: 'new-addon-slug' };
+      root.setProps({ addon: newAddon });
+
+      sinon.assert.calledWith(fakeDispatch, fetchOtherAddonsByAuthors({
+        addonType: newAddon.type,
+        authors: newAddon.authors.map((author) => author.username),
+        errorHandlerId: createStubErrorHandler().id,
+        slug: newAddon.slug,
+      }));
+    });
+
+    it('does not fetch the other add-ons when they are already loaded', () => {
+      const fakeDispatch = sinon.stub();
+      const root = shallowRender({ addon: fakeAddon, dispatch: fakeDispatch });
+      fakeDispatch.reset();
+
+      const newAddon = { ...fakeAddon, slug: 'new-addon-slug' };
+      // These `addonsByAuthors` are related to the `newAddon` above, and it
+      // simulates the case when user has previously browsed this `newAddon`
```

Comment:
I know I'm nitpicking but <code>had previously visited the `newAddon` page, navigated away, and then visited the page again</code> would be clearer I think 😄